src/vs/platform/secrets/common/secrets.ts

```diff
@@ -19,6 +19,7 @@ export interface ISecretStorageProvider {
 	get(key: string): Promise<string | undefined>;
 	set(key: string, value: string): Promise<void>;
 	delete(key: string): Promise<void>;
+	keys(): Promise<string[]>;
```

Comment:
This has to be optional to start... because vscode.dev, github.dev, and maybe `code serve-web` & `@vscode/test-web` implement one of these providers and bring them in via:
https://github.com/microsoft/vscode/blob/73fbb0cccd17f892e8ade51199f7476ddd760ee1/src/vs/workbench/browser/web.api.ts#L237

since they don't have an implementation today, we should assume that this `keys` is optional... and error accordingly.

Reply:
edit: I don't think we need to worry about `@vscode/test-web`:
https://github.com/microsoft/vscode-test-web/blob/main/src/browser/main.ts#L266-L269

Reply:
I've made the change for vscode.dev so that should work